### ISSUE TITLE
Working with localstorage

### ISSUE THREAD

**Issue by crippledjosh** [wojtkowiak/meteor-desktop#63]
I don't know if this is the best place to ask this, but essentially I'm really struggling to use Ground:db within meteor-desktop, because it uses local storage which uses the url including the port number to specify local storage locations. meteor-desktop (potentially just as part of what whats copied from the cordova system) seems to be using some sort of local server or proxy, I can see it's assigning its own port number at  "[localServer] assigned port ****" this means that every time i load up my application it gets a randomish port so it may or may not pick up the correct local storage. Is there a way to force it to use a particular port for the local server or some other work around you might have thought of?

**Comment by wojtkowiak** [comment]
Hey,
basically, the port is random to prevent conflicts with other `meteor-desktop` apps running on the same desktop.
I know that problem because I needed to tackle it already. That is why I have made my own simple implementation https://github.com/wojtkowiak/meteor-desktop-localstorage but that will not help you in any way.
I will try to make the built-in localStorage work in `0.6.0` - I have a mechanism in plan that will just rename the localStorage file to the correct name (with the right port).
I will also probably add an option to set the port manually and disable auto port finding.

**Comment by crippledjosh** [comment]
Hi, 

Ahhh that makes sense. You've probably already thought of this, but if you're worried about conflicts just with other meteor-desktop apps, is there any way to have a random port for the first time its loaded, but then always use that same port from then on?

Yeah i assumed you probably had encountered it after i saw that you were working on a fully offline version. Yeah i had thought that could be an idea, to just change the file name. By right port you mean like 3000 or rename it every time you load up to the latest port number? Surely if you're naming it to 3000 that still won't solve the issue that local storage will only be looking for the currently assigned port?

Anyway thanks for getting back to me and keep up the great work.

**Comment by wojtkowiak** [comment]
![bez tytulu](https://cloud.githubusercontent.com/assets/8668306/23166996/1d27ad8a-f843-11e6-9c4b-29c44c619ff0.png)
it now looks like this - every time the app start the port is different so there is an empty localStorage created. I will try to rename the files before running Chrome so that it would not loose the previously saved storage. I will see how hacky and reliable it would be.

**Comment by wojtkowiak** [comment]
Tomorrow I will release a version with an experimental support for localStorage

**Comment by wojtkowiak** [comment]
Please check `0.6.0` with `"experimentalLocalStorage": true" in `settings.json`.
https://github.com/wojtkowiak/meteor-desktop/blob/master/CHANGELOG.md#v060-27022017

**Comment by crippledjosh** [comment]
It appears to be working, I think I've got an issue on my end I need to identify before I can be sure. Ground DB doesn't seem to be grounding one of my collections and I'm not sure why, but I don't believe its related to meteor-desktop as my other collections are being grounded and are there on reload (the only reason I'm even suspicious of meteor-desktop is because before on the previous patch i was getting my collection when i hit the right port), but I'll try to sort it and then let you know how I get on.

**Comment by sferoze** [comment]
@wojtkowiak I added `"experimentalLocalStorage": true` in settings.json in the .desktop folder. The logs show that `experimentalLocalStorage` is enabled but I do not notice any difference. 

Every time I open the desktop app, it is reading from a new local storage. If I keep opening and closing the app, sometimes it will read from a previously created local storage but most of the time it just creates a new local storage. 

so basically adding `experimentalLocalStorage: true` is not doing anything in my desktop app. 

I am too using Ground:db which also uses localforage. localforage uses either IndexedDB, WebSQL, or localstorage. I have tried forcing ground:db to use all 3 including the default localstorage but still the same issues. 

Each time the app starts, it creates a new storage, only sometimes using a previously created one.

**Comment by sferoze** [comment]
@crippledjosh any update with using ground:db with meteor desktop? It is not working for me. When you quit and reopen desktop app, new storage is created, only sometimes using a previously created storage.

**Comment by sferoze** [comment]
@wojtkowiak UPDATE:

So I am closer to understanding the issue.

Ground:DB is using IndexedDB for storing data offline. 

The new setting you added `experimentalLocalStorage: true` only works for data stored in localStorage. 

When I forced Ground:DB to use localStorage only, the key `experimentalLocalStorage: true` did seem to work to load the previously created localStorage for the app. 

But this is a problem if you want to use IndexedDB which is the preferred way to store offline data. 

Also Ground:DB uses `localStorage` instead of `Meteor._localStorage`. Do you see any issues with this?

**Comment by wojtkowiak** [comment]
I've responded here: https://github.com/wojtkowiak/meteor-desktop/issues/85#issuecomment-303398702

**Comment by wojtkowiak** [comment]
Hey, `1.0.0` is out that is now free from any `indexedDB` or `localStorage` issues.

**Comment by wojtkowiak** [comment]
`experimentalLocalStorage` is obsolete now (I forgot to mention that in changelog), you do not have to set anything, localstorage and indexedDB should work flawlessly out of the box without any switch.
